java-spiffe-core/src/main/java/io/spiffe/workloadapi/Address.java

```diff
@@ -0,0 +1,153 @@
+            throw new SocketEndpointAddressException(String.format("Workload endpoint socket URI must have a tcp:// or unix:// scheme: %s", address));
+        }
+
+        String error = null;
+        if (UNIX_SCHEME.equals(scheme)) {
+            error = validateUnixAddress(parsedAddress);
+        }
+
+        if (TCP_SCHEME.equals(scheme)) {
+            error = validateTcpAddress(parsedAddress);
+        }
+
+        if (StringUtils.isNotBlank(error)) {
+            throw new SocketEndpointAddressException(String.format(error, address));
+        }
+
+        return parsedAddress;
+    }
+
+    private static String validateUnixAddress(final URI parsedAddress) {
+        if (parsedAddress.isOpaque() && parsedAddress.isAbsolute()) {
+            return "Workload endpoint unix socket URI must not be opaque: %s";
+        }
+
+        if (StringUtils.isNotBlank(parsedAddress.getUserInfo())) {
+            return "Workload endpoint unix socket URI must not include user info: %s";
+        }
+
+        if (StringUtils.isNotBlank(parsedAddress.getRawQuery())) {
+            return "Workload endpoint unix socket URI must not include query values: %s";
+        }
+
+        if (StringUtils.isNotBlank(parsedAddress.getFragment())) {
+            return "Workload endpoint unix socket URI must not include a fragment: %s";
+        }
+        return "";
+    }
+
+    private static String validateTcpAddress(final URI parsedAddress) {
+        if (parsedAddress.isOpaque() && parsedAddress.isAbsolute()) {
+            return "Workload endpoint tcp socket URI must not be opaque: %s";
+        }
+
+        if (StringUtils.isNotBlank(parsedAddress.getUserInfo())) {
+            return "Workload endpoint tcp socket URI must not include user info: %s";
+        }
+
+        if (StringUtils.isBlank(parsedAddress.getHost())) {
+            return "Workload endpoint tcp socket URI must include a host: %s";
+        }
+
+        if (StringUtils.isNotBlank(parsedAddress.getPath())) {
+            return "Workload endpoint tcp socket URI must not include a path: %s";
+        }
+
+        if (StringUtils.isNotBlank(parsedAddress.getRawQuery())) {
+            return "Workload endpoint tcp socket URI must not include query values: %s";
+        }
+
+        if (StringUtils.isNotBlank(parsedAddress.getFragment())) {
+            return "Workload endpoint tcp socket URI must not include a fragment: %s";
+        }
+
+        String ip = parseIp(parsedAddress.getHost());
+        if (StringUtils.isBlank(ip)) {
+            return "Workload endpoint tcp socket URI host component must be an IP:port: %s";
+        }
+
+        int port = parsedAddress.getPort();
+        if (port == -1) {
+            return "Workload endpoint tcp socket URI host component must include a port: %s";
+        }
+        return "";
+    }
+
+    private static boolean isValid(final String scheme) {
+        return StringUtils.isNotBlank(scheme) && VALID_SCHEMES.contains(scheme);
```

Comment:
`StringUtils.isNotBlank()` should be overkill here, can probably replace with `scheme != null`

Reply:
Done.